packages/backend-core/src/features/index.ts

```diff
@@ -1,79 +1,108 @@
 import env from "../environment"
 import * as context from "../context"
+import { cloneDeep } from "lodash"
 
-export * from "./installation"
-
-/**
- * Read the TENANT_FEATURE_FLAGS env var and return an array of features flags for each tenant.
- * The env var is formatted as:
- *  tenant1:feature1:feature2,tenant2:feature1
- */
-export function buildFeatureFlags() {
-  if (!env.TENANT_FEATURE_FLAGS) {
-    return
+class Flag<T> {
+  static withDefault<T>(value: T) {
+    return new Flag(value)
   }
 
-  const tenantFeatureFlags: Record<string, string[]> = {}
+  private constructor(public defaultValue: T) {}
+}
 
-  env.TENANT_FEATURE_FLAGS.split(",").forEach(tenantToFeatures => {
-    const [tenantId, ...features] = tenantToFeatures.split(":")
+// This is the primary source of truth for feature flags. If you want to add a
+// new flag, add it here and use the `fetch` and `get` functions to access it.
+// All of the machinery in this file is to make sure that flags have their
+// default values set correctly and their types flow through the system.
+const FLAGS = {
```

Comment:
One thing we should probably do - all these current flags are *very* old - I think we should probably remove them - we could instead use one of the features that is actually feature flagged right now instead, would it be worth considering moving SQS to use this mechanism instead?

All the feature flags that are here can 100% be removed as they should never be disabled now - the features are ancient, we've just never removed them!

Reply:
Happy for me to do this in a follow-up PR?

Reply:
Yeap sounds like a plan!